### ISSUE TITLE
Sidebar menu child items can't be dismissed via keyboard

### ISSUE THREAD

**Issue by SteelWagstaff** [pressbooks/pressbooks#4497]
Pressbooks sidebar menu implementation includes several child menu items. When focus has been shifted to this menu, users cannot exit the content via keyboard. Typically they should be able to press 'Esc' or similar action to have the menu dismissed. 

Our latest VPAT audit found that "it is not possible to dismiss the flyout main navigation menus without moving keyboard focus, and the menu obscures other content."

We should check to see if we inherit this pattern from core WordPress and if so, whether we're modifying the default behaviour in any wayt that reduces the accessibility of this submenu pattern.

<img width="512" height="411" alt="Image" src="https://github.com/user-attachments/assets/f3e701c1-3927-4366-8410-e3d8d5498ef5" />
